test_jpeg/jpegenc/subblocks/RLE/RLECore/rlecore.py

```diff
@@ -0,0 +1,179 @@
+""" core of the run length encoder module """
+
+from myhdl import always_comb, always_seq, block
+from myhdl import intbv, ResetSignal, Signal
+from myhdl.conversion import analyze
+from jpegenc.subblocks.RLE.RLECore.entropycoder import entropycoder
+
+
+WIDTH_RAM_ADDRESS = 6
+WIDTH_RAM_DATA = 12
+SIZE = 4
+
+class DataStream(object):
+    """ Input data streams into Rle Core """
+    def __init__(self, width=WIDTH_RAM_DATA):
+        self.start = Signal(bool(0))
+        self.input_val = Signal(intbv(0)[width:].signed())
+
+class RLESymbols(object):
+    """RLE symbols generatred by RLE Core"""
+    def __init__(self, width=WIDTH_RAM_DATA, size=SIZE):
+        self.runlength = Signal(intbv(0)[size:])
+        self.size = Signal(intbv(0)[size:])
+        self.amplitude = Signal(intbv(0)[width:].signed())
+        self.dovalid = Signal(bool(0))
+
+class RLEConfig(object):
+    """RLE configuration Signals are added here"""
+    def __init__(self):
+        self.color_component = Signal(intbv(0)[3:])
+        self.read_addr = Signal(intbv(0)[6:])
+        self.sof = Signal(bool(0))
+
+def sub(num1, num2):
+    """subtractor for Difference Encoder"""
+    return num1 - num2
+
+@block
+def rle(reset, clock, datastream, rlesymbols, rleconfig):
+    """ Input and start Signal send using datastream
+        and rle symbols are captured
+    """
+    rlesymbols_temp = RLESymbols()
+
+    prev_dc_0, prev_dc_1, prev_dc_2, zrl_data_in = [Signal(intbv(0)[
+        (WIDTH_RAM_DATA):].signed()) for _ in range(4)]
+
+    accumulator = Signal(intbv(0)[(WIDTH_RAM_DATA + 1):].signed())
+
+    read_en, divalid, divalid_en, zrl_processing = [
+        Signal(bool(0)) for _ in range(4)]
+
+    zero_cnt, read_cnt = [Signal(intbv(0)[7:]) for _ in range(2)]
+
+    write_cnt = Signal(intbv(0)[7:])
+
+    @always_comb
+    def assign():
+        """Assign outputs"""
+        rlesymbols.size.next = rlesymbols_temp.size
+        rlesymbols.amplitude.next = rlesymbols_temp.amplitude
+        rleconfig.read_addr.next = read_cnt
+
+
+    @always_seq(clock.posedge, reset=reset)
+    def mainprocessing():
+        """main processing"""
+        rlesymbols_temp.dovalid.next = 0
+        rlesymbols_temp.runlength.next = 0
+        rlesymbols.runlength.next = rlesymbols_temp.runlength
+        rlesymbols.dovalid.next = rlesymbols_temp.dovalid
+        divalid.next = read_en
+
+        if datastream.start == 1:
+            read_cnt.next = 0
+            read_en.next = 1
+            divalid_en.next = 1
+
+        if divalid == 1 and write_cnt == 63:
```

Comment:
`divalid` is a signal of type intbv. Number 63 indicates that we have processed 64 pixels in a block. I will make it more readable as you suggested